frappe/core/doctype/user_permission/test_user_permission.py

```diff
@@ -11,40 +11,55 @@ class TestUserPermission(unittest.TestCase):
 	def test_apply_to_all(self):
 		''' Create User permission for User having access to all applicable Doctypes'''
 		user = get_user()
-		param = get_params(user, apply = 1)
-		created = add_user_permissions(param)
-		self.assertEquals(created, 1)
+		param = get_params(user, apply_to_all = 1)
+		is_created = add_user_permissions(param)
+		self.assertEquals(is_created, 1)
+
+	def test_for_apply_to_all_on_update_from_apply_all(self):
```

Comment:
The test should not be dependent on other tests.

This test should first create a user permission doc with apply_to_all. Then try to update it again to check if it's not created again.

Reply:
refer line 22 and 24.
22 line creates if not exists and line 24 update it again. well both task is well handled by same function.